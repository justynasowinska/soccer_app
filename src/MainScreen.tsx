import React from 'react';
import { View, StyleSheet, SafeAreaView } from 'react-native';
import { SessionButton } from './components/SessionButton';
import { SoccerBoard } from './components/SoccerBoard';
import { SoccerSlider } from './components/SoccerSlider';
import { TOTAL_TIME } from './utils/dataUtils';
import { colors } from './utils/colors';

interface StateType {
    isPaused: boolean;
    currentTime: number;
    sliderValue: number;
}

export class MainScreen extends React.Component<{}, StateType> {
    clearInterval = 0;

    state: StateType = {
        isPaused: true,
        currentTime: 0,
        sliderValue: 0
    };

    componentWillUnmount() {
        clearInterval(this.clearInterval);
    }

    render() {
        const { isPaused, sliderValue, currentTime } = this.state;

        return (
            <View style={styles.container}>
                <SafeAreaView style={styles.safeAreaContainer}>
                    <SessionButton
                        isPaused={isPaused}
                        onPress={this.toggleSession}
                        containerStyles={styles.buttonContainer}
                    />
                    <SoccerBoard currentTime={currentTime} />
                    <SoccerSlider
                        value={sliderValue}
                        currentTime={this.state.currentTime}
                        totalTime={TOTAL_TIME}
                        onValueChange={this.onSliderValueChange}
                    />
                </SafeAreaView>
            </View>
        );
    }

    toggleSession = () => {
        const { isPaused } = this.state;

        if (isPaused) {
            this.runTimer();
        } else {
            this.pauseTimer();
        }

        this.setState({
            isPaused: !this.state.isPaused
        });
    }

    onSliderValueChange = (value: number) => {
        this.pauseTimer();

        this.setState({
            currentTime: value,
            isPaused: true
        });
    }

    runTimer = () => {
        this.clearInterval = setInterval(() => {
            const newTime = this.state.currentTime + 100;

            this.setState({
                currentTime: newTime,
                sliderValue: newTime
            });
        }, 100);
    }

    pauseTimer = () => {
        clearInterval(this.clearInterval);
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.mainDark
    },
    safeAreaContainer: {
        flex: 1,
        justifyContent: 'space-between',
        alignItems: 'center',
        padding: 40
    },
    buttonContainer: {
        alignSelf: 'flex-start',
    }
});