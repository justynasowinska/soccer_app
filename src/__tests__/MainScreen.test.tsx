import * as React from 'react';
import { shallow } from 'enzyme';

import { MainScreen } from '../MainScreen';

describe('MainScreen', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <MainScreen />
        );

        expect(component).toMatchSnapshot();
    });

    // TODO: Add more tests
});