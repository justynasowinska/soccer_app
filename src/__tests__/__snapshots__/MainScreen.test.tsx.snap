// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen matches to the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393551",
      "flex": 1,
    }
  }
>
  <SafeAreaView
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 40,
      }
    }
  >
    <SessionButton
      containerStyles={
        Object {
          "alignSelf": "flex-start",
        }
      }
      isPaused={true}
      onPress={[Function]}
    />
    <SoccerBoard
      currentTime={0}
    />
    <SoccerSlider
      currentTime={0}
      onValueChange={[Function]}
      totalTime={297800}
      value={0}
    />
  </SafeAreaView>
</View>
`;
